docs/developers_guide/Components.md

```diff
@@ -0,0 +1,74 @@
+# ComponentProvider API
+
+# Components implementation
+                 
+Component is defined by two parts:
+- specification
+- implementation
+
+Specification defines what are parameters of the component, and what are the results - i.e. how it transforms
+the Context. 
+                                 
+For simple components, the specification is derived from implementing method with the help of annotations. 
+
+## Specification
+                                                                                             
+### @MethodToInvoke
```

Comment:
maybe lets give here links to our sample implementations

Reply:
I added a few links